container/fs/fs_test.go

```diff
@@ -0,0 +1,281 @@
+	s.writeFile(filepath.Join("/tmp/proc", "diskstat_invalid_2"), []byte(invalidContentsParse))
+	s.writeFile(filepath.Join("/tmp/overlay/27fa0900fe22", "diskstats"), []byte(overlayContents))
+	s.writeFile(filepath.Join("/tmp/aufs/diff/27fa0900fe22", "diskstats"), []byte(aufsContents))
+	buf := new(syscall.Stat_t)
+	err = syscall.Stat("/tmp/test", buf)
+	if err != nil {
+		s.T().Fatal(err)
+	}
+	mockValidMount := mount.Info{
+		ID:         250,
+		Parent:     0,
+		Major:      int(major(buf.Dev)),
+		Minor:      int(minor(buf.Dev)),
+		Root:       "/tmp/test",
+		Mountpoint: "/",
+		Opts:       "rw,relatime",
+		Optional:   "shared:1",
+		Fstype:     "ext4",
+		Source:     "/tmp/test/dev/sdf100",
+		VfsOpts:    "rw,errors=remount-ro,data=ordered",
+	}
+	mockedMounts = append(mockedMounts, &mockValidMount)
+}
+
+func (s *FsSuite) TearDownSuite() {
+	err := os.RemoveAll("/tmp/aufs/")
+	if err != nil {
+		s.T().Fatal(err)
+	}
+	err = os.RemoveAll("/tmp/overlay/")
+	if err != nil {
+		s.T().Fatal(err)
+	}
+	err = os.RemoveAll("/tmp/proc/")
+	if err != nil {
+		s.T().Fatal(err)
+	}
+	err = os.RemoveAll("/tmp/var/")
+	if err != nil {
+		s.T().Fatal(err)
+	}
+	err = os.RemoveAll("/tmp/test/")
+	if err != nil {
+		s.T().Fatal(err)
+	}
+}
+
+func (s *FsSuite) TestFS() {
+	Convey("Check usage parser valid options", s.T(), func() {
+		usage, err := diskUsage("cat", []string{"/tmp/test/usage"})
+		So(usage, ShouldEqual, 6379736)
+		So(err, ShouldBeNil)
+	})
+	Convey("Check usage parser invalid options", s.T(), func() {
+		usage, err := diskUsage("cat", []string{"/tmp/test/usage_invalid"})
+		So(usage, ShouldEqual, 0)
+		So(err, ShouldNotBeNil)
+	})
+	Convey("Check usage parser invalid data", s.T(), func() {
+		usage, err := diskUsage("cat", []string{"/tmp/test/usage_invalid_parse"})
+		So(usage, ShouldEqual, 0)
+		So(err, ShouldNotBeNil)
+	})
+	Convey("Check getDiskStatsMap", s.T(), func() {
+		dsm, err := getDiskStatsMap("/tmp/proc/diskstats")
+		So(dsm[DeviceId{Major: 8, Minor: 1}].ReadsCompleted, ShouldEqual, 40)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].ReadsMerged, ShouldEqual, 0)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].SectorsRead, ShouldEqual, 280)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].ReadTime, ShouldEqual, 223)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].WritesCompleted, ShouldEqual, 7)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].WritesMerged, ShouldEqual, 0)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].SectorsWritten, ShouldEqual, 22)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].WriteTime, ShouldEqual, 108)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].IoInProgress, ShouldEqual, 0)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].IoTime, ShouldEqual, 330)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].WeightedIoTime, ShouldEqual, 330)
+		So(err, ShouldBeNil)
+	})
+	Convey("Check getDiskStatsMap no file", s.T(), func() {
+		_, err := getDiskStatsMap("/tmp/proc/diskstat")
+		So(err, ShouldBeNil)
+	})
+	Convey("Check getDiskStatsMap no file", s.T(), func() {
```

Comment:
Please change titles of two "Convey"s, now you have the same title for three "Convey"s.